test/extended/cli/mustgather.go

```diff
@@ -173,81 +173,94 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
-				if (strings.Contains(path, "-termination-") && strings.HasSuffix(path, ".log.gz")) || strings.HasSuffix(path, "termination.log.gz") {
-					// these are expected, but have unstructured log format
-					return nil
-				}
-
-				isAuditFile := (strings.Contains(path, "-audit-") && strings.HasSuffix(path, ".log.gz")) || strings.HasSuffix(path, "audit.log.gz")
-				o.Expect(isAuditFile).To(o.BeTrue())
+		// for some crazy reason, it seems that the files from must-gather take time to appear on disk for reading.  I don't understand why
+		// but this was in a previous commit and I don't want to immediately flake: https://github.com/openshift/origin/commit/006745a535848e84dcbcdd1c83ae86deddd3a229#diff-ad1c47fa4213de16d8b3237df5d71724R168
+		// so we're going to try to get a pass every 10 seconds for a minute.  If we pass, great.  If we don't, we report the
+		// last error we had.
+		var lastErr error
+		err = wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
+			// make sure we do not log OAuth tokens
+			for auditDirectory, expectedNumberOfAuditEntries := range expectedDirectoriesToExpectedCount {
+				eventsChecked := 0
+				err := filepath.Walk(auditDirectory, func(path string, info os.FileInfo, err error) error {
+					g.By(path)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					if info.IsDir() {
+						return nil
+					}
 
-				// at this point, we expect only audit files with json events, one per line
+					if (strings.Contains(path, "-termination-") && strings.HasSuffix(path, ".log.gz")) || strings.HasSuffix(path, "termination.log.gz") {
+						// these are expected, but have unstructured log format
+						return nil
+					}
 
-				readFile := false
+					isAuditFile := (strings.Contains(path, "-audit-") && strings.HasSuffix(path, ".log.gz")) || strings.HasSuffix(path, "audit.log.gz")
+					o.Expect(isAuditFile).To(o.BeTrue())
 
-				file, err := os.Open(path)
-				o.Expect(err).NotTo(o.HaveOccurred())
-				defer file.Close()
+					// at this point, we expect only audit files with json events, one per line
 
-				fi, err := file.Stat()
-				o.Expect(err).NotTo(o.HaveOccurred())
+					readFile := false
 
-				// it will happen that the audit files are sometimes empty, we can
-				// safely ignore these files since they don't provide valuable information
-				if fi.Size() == 0 {
-					return nil
-				}
+					file, err := os.Open(path)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					defer file.Close()
 
-				gzipReader, err := gzip.NewReader(file)
-				o.Expect(err).NotTo(o.HaveOccurred())
+					fi, err := file.Stat()
+					o.Expect(err).NotTo(o.HaveOccurred())
 
-				scanner := bufio.NewScanner(gzipReader)
-				for scanner.Scan() {
-					text := scanner.Text()
-					if !strings.HasSuffix(text, "}") {
-						continue // ignore truncated data
+					// it will happen that the audit files are sometimes empty, we can
+					// safely ignore these files since they don't provide valuable information
+					if fi.Size() == 0 {
```

Comment:
because no events within that window?

Reply:
> because no events within that window?

pre-existing. I don't know yet.  I think it's related to the whacky code from before that waited 10 seconds

Reply:
oh wait, it's because their control plane is unreachable from pods, so audit logs are unreachable